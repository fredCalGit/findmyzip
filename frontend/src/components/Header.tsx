import React, { useContext } from "react";
import { Brightness4, Brightness7 } from "@mui/icons-material";
import { CardHeader, IconButton, useTheme } from "@mui/material";
import { ColorModeContext } from "../App";

export const Header = () => {
  const theme = useTheme();
  const colorMode = useContext(ColorModeContext);

  return (
    <CardHeader
      title="whereZip"
      action={
        <IconButton
          sx={{ ml: 1 }}
          onClick={colorMode.toggleColorMode}
          color="inherit"
        >
          {theme.palette.mode === "dark" ? <Brightness7 /> : <Brightness4 />}
        </IconButton>
      }
    ></CardHeader>
  );
};
