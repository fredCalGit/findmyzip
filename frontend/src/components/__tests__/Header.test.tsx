import React from "react";
import { render } from "@testing-library/react";
import { Header } from "../Header";

test("renders the Header", () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});
