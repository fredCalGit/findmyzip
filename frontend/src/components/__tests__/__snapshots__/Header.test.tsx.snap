// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header 1`] = `
<div>
  <div
    class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
      >
        findMyZip
      </span>
    </div>
    <div
      class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="Brightness4Icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
